src/core/execute.c

```diff
@@ -4878,6 +4885,22 @@ int exec_context_get_clean_mask(ExecContext *c, ExecCleanMask *ret) {
         return 0;
 }
 
+int exec_context_cpu_affinity_from_numa(ExecContext *c) {
+        assert(c);
+
+        if (!c->numa_policy.derive_cpu_affinity_from_numa_mask)
+                return 0;
+
+        if (!c->numa_policy.nodes.set) {
+                log_warning_errno(SYNTHETIC_ERRNO(EINVAL), "Can't derive CPU affinity mask from NUMA mask because NUMA mask is not set, ignoring");
+                return 0;
+        }
+
+        (void) cpu_set_reset(&c->cpu_set);
+
+        return numa_to_cpu_set(&c->numa_policy, &c->cpu_set);
```

Comment:
don't modify ExecContext in this function, just return what you generate. ExecContext should stay unmodified